lib/ask/runtime/questionnaire_simulator.ex

```diff
@@ -139,12 +169,16 @@ defmodule Ask.Runtime.QuestionnaireSimulator do
 
   defp handle_app_reply(simulation, respondent, reply, status) do
     messages = simulation.messages ++ AOMessage.create_all(reply)
+    # TODO: support explanations in mobileweb
     submitted_steps = simulation.submissions ++ SubmittedStep.build_from_responses(respondent, simulation) ++ SubmittedStep.new_explanations(reply)
     current_step = current_step(reply)
 
-    QuestionnaireSimulatorStore.add_respondent_simulation(respondent.id, %Ask.QuestionnaireSimulation{simulation | respondent: sync_respondent(respondent), messages: messages, submissions: submitted_steps})
-    |> QuestionnaireSimulationStep.sync_build(current_step, status)
-    |> Response.success
+    simulation = %{simulation | respondent: sync_respondent(respondent), messages: messages, submissions: submitted_steps}
```

Comment:
I'm rusty with Elixir, but shouldn't be "safer" this struct to be `%Ask.QuestionnaireSimulation{simulation | ...}` as it was before?